tests/testing_harness.py

```diff
@@ -0,0 +1,116 @@
+from __future__ import print_function
+
+import filecmp
+import glob
+from optparse import OptionParser
+import os
+from subprocess import Popen, STDOUT, PIPE, call
+import sys
+
+import numpy as np
+
+sys.path.insert(0, '../..')
+from openmc.statepoint import StatePoint
+
+
+class TestHarness():
+    def __init__(self, statepoint_name, tallies_present=False):
+        self._sp_name = statepoint_name
+        self._tallies = tallies_present
+        self._opts = None
+        self._args = None
+
+
+    def execute_test(self):
+        self._parse_args()
+        try:
+            self._run_openmc()
+            self._test_output_created()
+            self._get_results()
+            self._compare_results()
+        finally:
+            self._cleanup()
+
+
+    def _parse_args(self):
+        parser = OptionParser()
+        parser.add_option('--mpi_exec', dest='mpi_exec', default='')
+        parser.add_option('--mpi_np', dest='mpi_np', default='3')
+        parser.add_option('--exe', dest='exe')
+        (self._opts, self._args) = parser.parse_args()
+        if self._opts.exe is None:
+            raise Exception('Must specify OpenMC executable from command line '
+                            'with --exe.')
+
+
+    def _run_openmc(self):
+        if self._opts.mpi_exec != '':
```

Comment:
Why not use the ``openmoc.Executor`` class here and elsewhere where you run OpenMC? It should further simplify the code and allow for greater code reuse (though you will need to modify ``Executor`` to be able to set the executable path).